appIndicator.js

```diff
@@ -724,11 +724,28 @@ class AppIndicatorsIconActor extends St.Icon {
             icon = this._indicator.overlayIcon;
             break;
         }
+        
+        let name_alt = null;
+        for (let cust_ics of this._custom_icon_array) {
+            if (this._indicator.id == cust_ics[0]){
+                name_alt = cust_ics[1]
+                }
+        }
 
         const [name, pixmap, theme] = icon;
         let gicon = null;
         try {
-            if (name && name.length) {
+            if ((name && name.length) && (name_alt && name_alt.length)) {
+        	gicon = await this._cacheOrCreateIconByName(iconSize, name_alt, theme);
+        	if (!gicon){
+        		gicon = await this._cacheOrCreateIconByName(iconSize, name, theme);}
+                if (!gicon && pixmap)
+                    gicon = await this._createIconFromPixmap(iconSize, pixmap, iconType);
+            } else if ((pixmap && pixmap.length) && (name_alt && name_alt.length)) {
+                gicon = await this._cacheOrCreateIconByName(iconSize, name_alt, theme);
+                if (!gicon)
+                    gicon = await this._createIconFromPixmap(iconSize, pixmap, iconType);   
+            } else if (name && name.length) {
```

Comment:
IIRC this is not needed, as it would be handled below anyways.

Reply:
Mh, I see, well I think in general this whole set of change could be changed by using a function that instead, given the named icon or pixmap we've, we returns a custom icon name if any, we try to load it and on failure we'd just fallback to default.

Reply:
Also probably we can easily support using alternate icon paths, not only icon-names, so if the alternate name is actually a path (has to start with `/` or maybe even `file://`) we can just load from that.

Reply:
I moved the stuff into its own function. The thing with the files instead of icon names will actually already work, since there was already a hack for that in the original code (presumably for indicator-multiload)